### PR TITLE
Search component template update

### PR DESCRIPTION
**Pull request by rayto22**
Updated template of search component and fixed small lint errors.
image
Also created a tests for method getShelters of shelterService.

### CODE REVIEW

File: src/app/shelters/shelters-service/shelters.service.spec.ts [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  it('should get allShelters', (done: DoneFn) => {
+    const configUrl = {
+      "sheltersApi": "https://familynetserver.azurewebsites.net/api/v1/childrenHouse"
+    }
+    const mockShelters = [
+      {"id":53,"name":"Ромашка","adressID":114,"rating":11.0,"avatar":null,"locationID":null,"photoPath":"https://familynetserver.azurewebsites.net/3nf3pby0.jpg"}
+    ];
+
+    configServiceSpy.getConfig.and.returnValue(of(configUrl).pipe(take(1)));
+    httpClientSpy.get.and.returnValue([mockShelters]);
```
Comment (ksenya-ta) [inline src/app/shelters/shelters-service/shelters.service.spec.ts]: should be observable?

File: src/app/shelters/shelters-service/shelters.service.spec.ts [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+    service.getShelters().subscribe((shelters: Shelter[]) => {
+      expect(shelters).toEqual(mockShelters);
+      done();
+    });
+
+  });
+
+  // xit('should get filtered shelter list', () => {
```
Comment (ksenya-ta) [inline src/app/shelters/shelters-service/shelters.service.spec.ts]: do you still need it?

File: src/app/shelters/shelters-service/shelters.service.spec.ts (hunk 1 above)
Comment (rayto22) [reply src/app/shelters/shelters-service/shelters.service.spec.ts]: Im deeply sorry, on that time i experimented with tests, and for communication of my home PC and notebook i pushed this unfinished variant to this branch. That implemantation of tests do not  conform to standards for reviewing, my mistake. I should not have pushed it here in the first place. For now i added updated variant for reviewing.

File: src/app/shelters/shelters-service/shelters.service.spec.ts (hunk 2 above)
Comment (rayto22) [reply src/app/shelters/shelters-service/shelters.service.spec.ts]: Unneeded tests were removed.

File: src/app/shelters/shelters-service/shelters.service.spec.ts [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    const configUrl = {
+      "sheltersApi": "https://familynetserver.azurewebsites.net/api/v1/childrenHouse"
+    };
+    const mockShelters = [
+      {"id":14,"name":"Золушка","adressID":115,"rating":11.0,"avatar":null,"locationID":null,"photoPath":"https://familynetserver.azurewebsites.net/3nf3123pby0.jpg"},
+      {"id":54,"name":"Золушка","adressID":115,"rating":11.0,"avatar":null,"locationID":null,"photoPath":"https://familynetserver.azurewebsites.net/3nf3123pby0.jpg"}
+    ];
+
+    configServiceSpy.getConfig.and.returnValue(of(configUrl));
+    httpClientSpy.get.and.returnValue(of(mockShelters));
```
Comment (ksenya-ta) [inline src/app/shelters/shelters-service/shelters.service.spec.ts]: when you need the same setup for a group of tests you can group them in one `describe `and add it in the `beforeEach `section

File: src/app/shelters/shelters-service/shelters.service.spec.ts [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+    service.getShelters({name: saerchValue}).subscribe((shelters: Shelter[]) => {
+      expect(httpClientSpy.get).toHaveBeenCalledWith(configUrl.sheltersApi, { params });
+      expect(shelters).toEqual(mockShelters);
```
Comment (ksenya-ta) [inline src/app/shelters/shelters-service/shelters.service.spec.ts]: if you need to test that this method sends correct http params then just check it, no need to also compare the returned data, if you expect to test that returned from API call data was processed it is better to create separate test for it

**Pull request merged**